### PR TITLE
Prevent Pagination Error

### PR DESCRIPTION
**Pull request by PhoenixPeca**
Hi,
I have discovered some bugs and implemented a fix through this PR.
**PR Summary:**
- added if page number is more than 18 digits, return a 404 error, instead of 500 or 422 errors

BUGS:
> Ghost shows a 500 error to modified page numbers within the range of 19-20 digits
> 
> capture
> here's an example: https://blog.ghost.org/page/9999999999999999999/

> Ghost shows a 422 error to modified page numbers with more than 20 digits
> 
> capture
> here's an example: https://blog.ghost.org/page/999999999999999999999/

### CODE REVIEW

**Comment by PhoenixPeca** [comment]
BTW, what's up with the cute emojis on the title of other PRs?

**Comment by PhoenixPeca** [comment]
Update:
I believe this is related to this integer: [9223372036854775807](https://en.wikipedia.org/wiki/9223372036854775807), the known limits of 64-bit systems.

I also believe that this is also related to this integer (but I still can't verify it): [2147483647](https://en.wikipedia.org/wiki/2147483647_(number)), the known limits of 32-bit systems.
----------------------------
This PR works, but is somehow a bit hacky, So I recommend adding a custom check instead.
In a nutshell, our best approach would be to add an if statement that would say something like:
> If (_[page-number-in-url]_ > _[total-number-of-pagination-pages]_) {
>           return 404 error;
> }

**Comment by kirrg001** [comment]
Hey @PhoenixPeca 👋 Thanks for your PR.

The error occurs because the database can't handle this big integer, as you pointed out.
`offset` is a unsigned big integer in MySQL with max length of 18446744073709551615 (64bit).
`offset` is a signed big integer in Sqlite with max length of 9223372036854775807 (64bit).

See https://dev.mysql.com/doc/refman/5.7/en/integer-types.html

There is the option to query the database before to see how many pages are available, but this would make the request slower and the use case is in general very raw. 

I wouldn't compare the length of the page, because this is not accurate e.g. we allow 10 digits, but 4294967296 has also 10 digits (max is4294967295 for 32-bit) .

I would rather catch the error and transform the error into a not found error.
Here: https://github.com/TryGhost/Ghost/blob/master/core/server/models/plugins/pagination.js#L197

**Review by kirrg001 — CHANGES_REQUESTED** [review]
https://github.com/TryGhost/Ghost/pull/8129#issuecomment-286061529

**Comment by ErisDS** [comment]
Going to close this for now as we are cleaning up post 1.0. Feel free to open a new PR with the recommended approach.
